extensions/cocostudio/timeline/CCSkeletonNode.js

```diff
@@ -0,0 +1,188 @@
+/****************************************************************************
+ Copyright (c) 2015-2016 Chukong Technologies Inc.
+
+ http://www.cocos2d-x.org
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+/**
+ * SkeletonNode
+ * base class
+ * @class
+ */
+ccs.SkeletonNode = (function(){
+
+    var BoneNode = ccs.BoneNode;
+
+    var type = {
+        p: cc.p,
+        size: cc.size,
+        rect: cc.rect
+    };
+
+    var SkeletonNode = BoneNode.extend(/** @lends ccs.SkeletonNode# */{
+        _subBonesMap: null,
+
+        _squareVertices: null,
+        _squareColors: null,
+        _noMVPVertices: null,
+        _suitMap: null,
+
+        _sortedAllBonesDirty: false,
+        _sortedAllBones: null,
+        _batchedBoneVetices: null,
+        _batchedBoneColors: null,
+        _batchedVeticesCount: null,
+        _batchBoneCommand: null,
+
+        ctor: function(){
+            this._squareVertices = [
+                {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0},
+                {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}
+            ];
+            BoneNode.prototype.ctor.call(this);
+            this._subBonesMap = {};
+
+            this._rackLength = this._rackWidth = 20;
+            this.setContentSize(cc.size(20, 20));
```

Comment:
the latest code remove set archorpoint and contentsize's value
cocos2d/cocos2d-x#12973